lib/reek/source/source_code.rb

```diff
@@ -37,9 +37,10 @@ def initialize(code, description, parser = Parser::Ruby22)
       # @return an instance of SourceCode
       def self.from(source)
         case source
-        when File   then new(source.read, source.path)
-        when IO     then new(source.readlines.join, 'STDIN')
-        when String then new(source, 'string')
+        when File     then new(source.read, source.path)
+        when IO       then new(source.readlines.join, 'STDIN')
+        when Pathname then new(source.read, source.to_s)
```

Comment:
Oh, `IO` objects, I can get behind that. `IO` protocol in Ruby is one of the parts I really like. :)